leoism/step/portfolio/src/main/webapp/script.js

```diff
@@ -66,21 +66,32 @@ function createComment(commentInformation) {
 function loadComments() {
   const responsePromise = fetch('/data');
   const commentsContainer = document.getElementById('comments-container');
-  const amtCurrentlyDisplayed = parseInt(commentsContainer.getAttribute('data-display'));
+  const amtCurrentlyDisplayed =
+      parseInt(commentsContainer.getAttribute('data-display'));
 
   const nextTenComments = amtCurrentlyDisplayed + 10;
+
+  // Remains true if the next iteration of 10 comments included 10 comments.
+  let nextTenCommentsLoaded = true;
   responsePromise.then((response) => response.json()).then((comments) => {
     for (let i = amtCurrentlyDisplayed; i < nextTenComments; i++) {
+      // Make load more button not appear when all comments have been loaded.
+      if (i === comments.length - 1) {
+        const loadMoreButton = document.getElementById('load-more-button');
+        loadMoreButton.style.display = 'none';
+      }
+
       if (comments[i] === undefined) {
```

Comment:
When would this be undefined? Does being undefined mean that we are at then end of the list?

Reply:
The for loop iterates 10 times every time. If the list is short of ten, say the list only has 4 elements, then comments[i] would be undefined at the 5th iteration. At this point, we know we iterated i times so the data-display attribute is updated to i

Reply:
I added a comment to explain this